### ISSUE TITLE
Upgrade to Django 4.2

### ISSUE THREAD

**Issue by matthewelwell** [SolidStateGroup/bullet-train-api#2088]
https://www.djangoproject.com/weblog/2023/apr/03/django-42-released/

Known TODOs:
 * Replace / remove third party pytest plugin with the one available in pytest-django [here](https://github.com/Flagsmith/flagsmith/blob/1aa00c97f0afea709436fac328e923bc1024740a/api/tests/unit/task_processor/test_unit_task_processor_decorators.py#L62-L63)

**Comment by oohsai** [comment]
Hey! I'd like to work over this issue. Does this include just upgrading django to 4.2 and running all the security checks afterwards if code working properly or not?

**Comment by dabeeeenster** [comment]
Yep - getting all the Github Action tests to pass!

**Comment by SD-13** [comment]
@matthewelwell @dabeeeenster I want to work on this. please let me if it's available. Thanks!

**Comment by matthewelwell** [comment]
Hi @SD-13 , I don't know if @oohsai is working on it already? If so, please assign to yourself. If not, @SD-13 feel free to take it.

**Comment by oohsai** [comment]
Hey @SD-13, Started Working on it but As I'm just startin off with open source. You can assign this issue to yourself. Just a humble request to drop on how you resolved this issue would be of great help.

**Comment by oohsai** [comment]
Hey @SD-13, Started Working on it but As I'm just startin off with open source. You can assign this issue to yourself. Just a humble request to drop on how you resolved this issue would be of great help.

**Comment by SD-13** [comment]
@oohsai sure thing 😃. @matthewelwell Can you please assign me. Thanks!

**Comment by gagantrivedi** [comment]
> @oohsai sure thing 😃. @matthewelwell Can you please assign me. Thanks!

Here you go!

**Comment by SD-13** [comment]
I am upgrading django versions incrementally by fixing errors in each version. As part of that, I upgraded django to 4.0 and django-trench from 0.2.3 to 0.3.0 and now I am getting this error
```
ImportError: cannot import name 'AuthTokenLoginOrRequestMFACode' from 'trench.views.authtoken'
```
After digging a bit I found that, the `AuthTokenLoginOrRequestMFACode` and `AuthTokenLoginWithMFACode` is removed in this PR. So what should we do in this case?

**Comment by gagantrivedi** [comment]
> I am upgrading django versions incrementally by fixing errors in each version. As part of that, I upgraded django to 4.0 and django-trench from 0.2.3 to 0.3.0 and now I am getting this error
> 
> ```
> ImportError: cannot import name 'AuthTokenLoginOrRequestMFACode' from 'trench.views.authtoken'
> ```
> 
> After digging a bit I found that, the `AuthTokenLoginOrRequestMFACode` and `AuthTokenLoginWithMFACode` is removed in this PR. So what should we do in this case?

@SD-13  can you add the PR in which it was removed?

**Comment by SD-13** [comment]
> @SD-13 can you add the PR in which it was removed?

@gagantrivedi https://github.com/merixstudio/django-trench/pull/115

For now, I am trying the changes in [this PR](https://github.com/SD-13/flagsmith/pull/5)

**Comment by matthewelwell** [comment]
This was previously blocked by the postgres engine upgrade and determining oracle support. These are now complete so we can work on this upgrade.

**Comment by matthewelwell** [comment]
This has been released in 2.138.2 🥳
